File: 2022/20221102/LeastCommonMultiple/src/test/java/LeastCommonMultipleTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class LeastCommonMultipleTest {
    @Test
    void solution() {
        LeastCommonMultiple test = new LeastCommonMultiple();
        assertEquals(168, test.solution(new int[]{2, 6, 8, 14}));
        assertEquals(6, test.solution(new int[]{1, 2, 3}));
        assertEquals(3, test.solution(new int[]{3}));
    }

    @Test
    void calcLeastCommonMultiple() {
        LeastCommonMultiple test = new LeastCommonMultiple();
        assertEquals(2, test.calcLeastCommonMultiple(1, 2));
        assertEquals(6, test.calcLeastCommonMultiple(2, 3));
    }

    @Test
    void calcGreatestCommonDivisor() {
        LeastCommonMultiple test = new LeastCommonMultiple();
        assertEquals(1, test.calcGreatestCommonDivisor(1, 2));
        assertEquals(2, test.calcGreatestCommonDivisor(2, 6));
    }
}

File: 2022/20221103/LeastCommonMultiple.test.js
import solution, { GCD, LCM } from './LeastCommonMultiple';

test('solution', () => {
  expect(solution([2, 6, 8, 14])).toBe(168);
  expect(solution([1, 2, 3])).toBe(6);
});

test('LCM', () => {
  expect(LCM(2, 3)).toBe(6);
  expect(LCM(3, 4)).toBe(12);
});

test('GCD', () => {
  expect(GCD(1, 2)).toBe(1);
  expect(GCD(2, 6)).toBe(2);
});
